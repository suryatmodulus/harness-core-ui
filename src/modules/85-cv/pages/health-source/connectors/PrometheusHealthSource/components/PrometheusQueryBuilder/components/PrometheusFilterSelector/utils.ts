/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@wings-software/uicore'

export function updateLabelOfSelectedFilter(
  selectedValue: string,
  item: MultiSelectOption,
  items: MultiSelectOption[]
): MultiSelectOption {
  const currentItem = items.find(listItem => listItem.value === item.value)
  if (currentItem) {
    const splitString = (currentItem.label as string).split(':')[0]
    return { ...currentItem, label: `${splitString}:${selectedValue}` }
  }

  return item
}

export function updateMultiSelectOption(
  newOption: MultiSelectOption,
  currentList: MultiSelectOption[]
): MultiSelectOption[] {
  if (!currentList?.length) {
    return []
  }

  const newOptionIndex = currentList.findIndex(listItem => listItem.value === newOption.value)
  if (newOptionIndex > -1) {
    currentList[newOptionIndex] = newOption
  }

  return Array.from(currentList)
}
