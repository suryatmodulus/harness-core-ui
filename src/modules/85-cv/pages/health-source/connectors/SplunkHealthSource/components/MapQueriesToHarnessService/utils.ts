/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import type { UseStringsReturn } from 'framework/strings'
import { initialFormData, MapSplunkToServiceFieldNames } from './constants'
import type { MapSplunkQueryToService } from './types'

type UpdateSelectedQueriesMap = {
  updatedMetric: string
  oldMetric: string
  mappedMetrics: Map<string, MapSplunkQueryToService>
  formikProps: FormikProps<MapSplunkQueryToService | undefined>
}

export function updateSelectedMetricsMap({
  updatedMetric,
  oldMetric,
  mappedMetrics,
  formikProps
}: UpdateSelectedQueriesMap): any {
  const updatedMap = new Map(mappedMetrics)

  // in the case where user updates query name, update the key for current value
  if (oldMetric !== formikProps.values?.metricName) {
    updatedMap.delete(oldMetric)
  }

  // if newly created query create form object
  if (!updatedMap.has(updatedMetric)) {
    updatedMap.set(updatedMetric, { ...initialFormData, metricName: updatedMetric })
  }

  // update map with current form data
  if (formikProps.values?.metricName) {
    updatedMap.set(formikProps.values.metricName, formikProps.values as MapSplunkQueryToService)
  }
  return { selectedMetric: updatedMetric, mappedMetrics: updatedMap }
}

export function validateMappings(
  getString: UseStringsReturn['getString'],
  createdMetrics: string[],
  selectedMetricIndex: number,
  values?: MapSplunkQueryToService
): { [fieldName: string]: string } {
  const requiredFieldErrors = {
    ...(!values?.metricName && {
      [MapSplunkToServiceFieldNames.METRIC_NAME]: getString('cv.monitoringSources.queryNameValidation')
    }),
    ...(!values?.query && {
      [MapSplunkToServiceFieldNames.QUERY]: getString('cv.monitoringSources.gco.manualInputQueryModal.validation.query')
    }),
    ...(!values?.serviceInstance && {
      [MapSplunkToServiceFieldNames.SERVICE_INSTANCE]: getString(
        'cv.monitoringSources.gcoLogs.validation.serviceInstance'
      )
    })
  }

  const duplicateNames = createdMetrics?.filter((metricName, index) => {
    if (index === selectedMetricIndex) {
      return false
    }
    return metricName === values?.metricName
  })

  if (values?.metricName && duplicateNames.length) {
    requiredFieldErrors[MapSplunkToServiceFieldNames.METRIC_NAME] = getString(
      'cv.monitoringSources.gcoLogs.validation.queryNameUnique'
    )
  }

  return requiredFieldErrors
}
