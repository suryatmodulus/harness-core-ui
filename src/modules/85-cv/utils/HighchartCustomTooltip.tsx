/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type {
  Chart as HighchartsChart,
  TooltipFormatterCallbackFunction,
  TooltipFormatterContextObject
} from 'highcharts'
import { useEffect, useState, useRef } from 'react'
import ReactDOM from 'react-dom'

const generateTooltipId = (chartId: number): string => `highcharts-custom-tooltip-${chartId}`

interface Props {
  chart: HighchartsChart | null
  children(formatterContext: TooltipFormatterContextObject): JSX.Element
}

export const HighchartCustomTooltip: React.FC<Props> = ({ chart, children }) => {
  const isInit = useRef(false)
  const [context, setContext] = useState<TooltipFormatterContextObject | null>(null)

  useEffect(() => {
    if (chart?.tooltip && chart.update) {
      const formatter: TooltipFormatterCallbackFunction = function () {
        // Ensures that tooltip DOM container is rendered before React portal is created.
        if (!isInit.current) {
          isInit.current = true

          chart.tooltip.refresh.apply(chart.tooltip, [this.point])
          chart.tooltip.hide(0)
        }

        setContext(this)

        return `<div id="${generateTooltipId(chart.index)}"></div>`
      }

      chart.update({
        tooltip: {
          formatter,
          useHTML: true
        }
      })
    }
  }, [chart])

  const node = chart?.index && document.getElementById(generateTooltipId(chart.index))

  return node && context ? ReactDOM.createPortal(children(context), node) : null
}
