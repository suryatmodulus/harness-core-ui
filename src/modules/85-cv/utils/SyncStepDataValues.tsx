/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useEffect } from 'react'
import pickBy from 'lodash-es/pickBy'
import merge from 'lodash-es/merge'

interface SyncStepDataValuesProps {
  values: any
  listenToValues: any
  onUpdate: (val: any) => void
}

export default function SyncStepDataValues({ values, listenToValues, onUpdate }: SyncStepDataValuesProps) {
  useEffect(() => {
    const update = pickBy({ ...listenToValues }, value => !!value)
    onUpdate(merge(values, update))
  }, [listenToValues])
  return null
}
