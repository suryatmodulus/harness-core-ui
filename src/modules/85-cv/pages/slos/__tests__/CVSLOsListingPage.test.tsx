import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, waitFor, screen } from '@testing-library/react'
import routes from '@common/RouteDefinitions'
import { TestWrapper } from '@common/utils/testUtils'
import * as cvServices from 'services/cv'
import CVSLOsListingPage from '../CVSLOsListingPage'
import type { CVSLOsListingPageProps, SLOCardHeaderProps } from '../CVSLOsListingPage.types'
import {
  testWrapperProps,
  pathParams,
  errorMessage,
  dashboardWidgetsResponse,
  dashboardWidgetsContent,
  userJourneyResponse
} from './CVSLOsListingPage.mock'

jest.mock('@cv/pages/slos/SLOCard/SLOCardHeader.tsx', () => ({
  __esModule: true,
  default: function SLOCardHeader({ onDelete }: SLOCardHeaderProps) {
    return (
      <span data-testid="slo-card-header">
        <button onClick={() => onDelete(dashboardWidgetsContent.sloIdentifier, dashboardWidgetsContent.title)}>
          Delete
        </button>
      </span>
    )
  }
}))

jest.mock('@cv/pages/slos/SLOCard/SLOCardContent.tsx', () => ({
  __esModule: true,
  default: function SLOCardContent() {
    return <span data-testid="slo-card-content" />
  }
}))

const ComponentWrapper: React.FC<CVSLOsListingPageProps> = ({ monitoredService }) => {
  return (
    <TestWrapper {...testWrapperProps}>
      <CVSLOsListingPage monitoredService={monitoredService} />
    </TestWrapper>
  )
}

describe('CVSLOsListingPage', () => {
  let useGetAllJourneys: jest.SpyInstance
  let useGetSLODashboardWidgets: jest.SpyInstance
  let useDeleteSLOData: jest.SpyInstance
  let refetchUserJourneys: jest.Mock
  let refetchDashboardWidgets: jest.Mock

  beforeEach(() => {
    refetchUserJourneys = jest.fn()
    refetchDashboardWidgets = jest.fn()
    useGetAllJourneys = jest.spyOn(cvServices, 'useGetAllJourneys').mockReturnValue({
      data: {},
      loading: false,
      error: null,
      refetch: jest.fn()
    } as any)

    useGetSLODashboardWidgets = jest.spyOn(cvServices, 'useGetSLODashboardWidgets').mockReturnValue({
      data: {},
      loading: false,
      error: null,
      refetch: refetchDashboardWidgets
    } as any)

    jest.spyOn(cvServices, 'useGetAllMonitoredServicesWithTimeSeriesHealthSources').mockReturnValue({
      data: [],
      loading: false,
      error: null,
      refetch: jest.fn()
    } as any)

    jest.spyOn(cvServices, 'useGetServiceLevelObjectivesRiskCount').mockReturnValue({
      data: {
        data: {
          riskCounts: [
            {
              displayName: 'Healthy',
              identifier: 'HEALTHY',
              count: 2
            }
          ],
          totalCount: 3
        }
      },
      loading: false,
      error: null,
      refetch: jest.fn()
    } as any)

    useDeleteSLOData = jest
      .spyOn(cvServices, 'useDeleteSLOData')
      .mockReturnValue({ mutate: jest.fn(), loading: false, error: null } as any)
  })

  test('Without monitoredServiceIdentifier it should render with the page header and +New SLO button', () => {
    render(<ComponentWrapper />)

    expect(screen.getByText('cv.slos.title')).toBeInTheDocument()
    expect(screen.getByText('cv.slos.newSLO')).toBeInTheDocument()
  })

  test('With monitoredServiceIdentifier it should not render with the page header and +New SLO button', () => {
    render(
      <ComponentWrapper
        monitoredService={{ identifier: 'monitored_service_identifier', name: 'monitored_service_identifier' }}
      />
    )

    expect(screen.queryByText('cv.slos.title')).not.toBeInTheDocument()
    expect(screen.queryByText('cv.slos.newSLO')).not.toBeInTheDocument()
  })

  test('add new SLO should go to create page', async () => {
    render(<ComponentWrapper />)

    userEvent.click(screen.getByText('cv.slos.newSLO'))

    expect(screen.getByText(routes.toCVCreateSLOs({ ...pathParams, module: 'cv' }))).toBeInTheDocument()
  })

  test('it should show the loader while fetching the user journeys', () => {
    useGetAllJourneys.mockReturnValue({ data: {}, loading: true, error: null, refetch: jest.fn() })

    render(<ComponentWrapper />)

    expect(screen.getByText('Loading, please wait...')).toBeInTheDocument()
    expect(screen.queryByText('First Journey')).not.toBeInTheDocument()
  })

  test('it should show the loader while fetching the dashboard widgets', () => {
    useGetSLODashboardWidgets.mockReturnValue({ data: {}, loading: true, error: null, refetch: jest.fn() })

    render(<ComponentWrapper />)

    expect(screen.getByText('Loading, please wait...')).toBeInTheDocument()
    expect(screen.queryByTestId('slo-card-container')).not.toBeInTheDocument()
  })

  test('it should show the loader while deleting a widget', () => {
    useGetAllJourneys.mockReturnValue({ data: userJourneyResponse, loading: false, error: null, refetch: jest.fn() })
    useDeleteSLOData.mockReturnValue({ mutate: jest.fn(), loading: true, error: null })
    useGetSLODashboardWidgets.mockReturnValue({
      data: dashboardWidgetsResponse,
      loading: false,
      error: null,
      refetch: jest.fn()
    })

    render(<ComponentWrapper />)

    expect(screen.getByText('Loading, please wait...')).toBeInTheDocument()
    expect(screen.getAllByTestId('slo-card-container')).toHaveLength(1)
  })

  test('page retry should trigger both dashboard widget and user journey APIs when both returned error response', () => {
    useGetAllJourneys.mockReturnValue({
      data: {},
      loading: false,
      error: { message: errorMessage },
      refetch: refetchUserJourneys
    })
    useGetSLODashboardWidgets.mockReturnValue({
      data: {},
      loading: false,
      error: { message: errorMessage },
      refetch: refetchDashboardWidgets
    })

    render(<ComponentWrapper />)

    const onRetryButton = screen.getByRole('button', { name: 'Retry' })

    expect(onRetryButton).toBeInTheDocument()
    expect(refetchUserJourneys).not.toHaveBeenCalled()
    expect(refetchDashboardWidgets).not.toHaveBeenCalled()

    userEvent.click(onRetryButton)

    expect(refetchUserJourneys).toHaveBeenCalled()
    expect(refetchDashboardWidgets).toHaveBeenCalled()
  })

  test('page retry should only trigger the user journey API when dashboard widget API returned success response', () => {
    useGetAllJourneys.mockReturnValue({
      data: {},
      loading: false,
      error: { message: errorMessage },
      refetch: refetchUserJourneys
    })

    render(<ComponentWrapper />)

    const onRetryButton = screen.getByRole('button', { name: 'Retry' })

    expect(onRetryButton).toBeInTheDocument()
    expect(refetchUserJourneys).not.toHaveBeenCalled()

    userEvent.click(onRetryButton)

    expect(refetchUserJourneys).toHaveBeenCalled()
    expect(refetchDashboardWidgets).not.toHaveBeenCalled()
  })

  test('page retry should only trigger the dashboard widget API when userJourney API returned success response', () => {
    useGetSLODashboardWidgets.mockReturnValue({
      data: {},
      loading: false,
      error: { message: errorMessage },
      refetch: refetchDashboardWidgets
    })

    render(<ComponentWrapper />)

    const onRetryButton = screen.getByRole('button', { name: 'Retry' })

    expect(onRetryButton).toBeInTheDocument()
    expect(refetchDashboardWidgets).not.toHaveBeenCalled()

    userEvent.click(onRetryButton)

    expect(refetchUserJourneys).not.toHaveBeenCalled()
    expect(refetchDashboardWidgets).toHaveBeenCalled()
  })

  test('it should render page body no data state only if dashboard widgets and selected user journey are empty', () => {
    render(<ComponentWrapper />)

    expect(screen.getByText('cv.slos.noData')).toBeInTheDocument()
    expect(screen.queryByText('First Journey')).not.toBeInTheDocument()
    expect(screen.queryByTestId('slo-card-container')).not.toBeInTheDocument()
  })

  test('it should only render dashboard widgets when user journeys are empty', () => {
    useGetSLODashboardWidgets.mockReturnValue({
      data: dashboardWidgetsResponse,
      loading: false,
      error: null,
      refetch: jest.fn()
    })

    render(<ComponentWrapper />)

    expect(screen.queryByText('First Journey')).not.toBeInTheDocument()
    expect(screen.getByTestId('slo-card-container')).toBeInTheDocument()
  })

  test('Risk filter select and deselect', async () => {
    useGetAllJourneys.mockReturnValue({ data: userJourneyResponse, loading: false, error: null, refetch: jest.fn() })
    useGetSLODashboardWidgets.mockReturnValue({
      data: dashboardWidgetsResponse,
      loading: false,
      error: null,
      refetch: jest.fn()
    })

    const { container } = render(<ComponentWrapper />)

    expect(container).toMatchSnapshot()

    expect(screen.getAllByTestId('slo-card-container')).toHaveLength(1)
    expect(screen.getAllByTestId('slo-card-header')).toHaveLength(1)
    expect(screen.getAllByTestId('slo-card-content')).toHaveLength(1)

    expect(screen.getByText('Healthy').parentElement).not.toHaveClass('Card--selected')

    userEvent.click(screen.getByText('Healthy'))

    expect(screen.getByText('Healthy').parentElement).toHaveClass('Card--selected')

    userEvent.click(screen.getByText('Healthy'))

    expect(screen.getByText('Healthy').parentElement).not.toHaveClass('Card--selected')
  })

  test('deleting a widget', async () => {
    const deleteMutate = jest.fn()
    const refetch = jest.fn()

    useGetAllJourneys.mockReturnValue({ data: userJourneyResponse, loading: false, error: null, refetch: jest.fn() })
    useDeleteSLOData.mockReturnValue({ mutate: deleteMutate, loading: false, error: null })
    useGetSLODashboardWidgets.mockReturnValue({
      data: dashboardWidgetsResponse,
      loading: false,
      error: null,
      refetch: refetch
    })

    render(<ComponentWrapper />)

    expect(screen.getByTestId('slo-card-container')).toBeInTheDocument()

    userEvent.click(screen.getByText('Delete'))

    expect(deleteMutate).toHaveBeenCalledWith(dashboardWidgetsContent.sloIdentifier)

    await waitFor(() => expect(refetch).toHaveBeenCalled())
    await waitFor(() => expect(screen.getByText('cv.slos.sloDeleted')).toBeInTheDocument())
  })

  describe('Filters', () => {
    test('should check whether all the filters are present', () => {
      useGetSLODashboardWidgets.mockReturnValue({
        data: dashboardWidgetsResponse,
        loading: false,
        error: null,
        refetch: jest.fn()
      })
      render(<ComponentWrapper />)

      const userJourneyFilter = screen.getByTestId('userJourney-filter')
      const monitoredServicesFilter = screen.getByTestId('monitoredServices-filter')
      const sloTargetAndBudgetFilter = screen.getByTestId('sloTargetAndBudget-filter')
      const sliTypeFilter = screen.getByTestId('sliType-filter')

      expect(userJourneyFilter).toBeInTheDocument()
      expect(monitoredServicesFilter).toBeInTheDocument()
      expect(sloTargetAndBudgetFilter).toBeInTheDocument()
      expect(sliTypeFilter).toBeInTheDocument()
    })
  })
})
