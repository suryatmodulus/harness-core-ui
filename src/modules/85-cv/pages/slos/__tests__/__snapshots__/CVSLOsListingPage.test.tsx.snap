// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVSLOsListingPage Risk filter select and deselect 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="PageHeader--breadcrumbsDiv"
      >
        <div
          class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/account_id/settings"
            >
              <span
                class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                icon="cog"
              >
                <svg
                  data-icon="cog"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    cog
                  </desc>
                  <path
                    d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                common.accountSettings
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <h2
        class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-weight-bold"
        level="2"
      >
        cv.slos.title
      </h2>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon Button--variation Button--variation-primary"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          cv.slos.newSLO
        </span>
      </button>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody pageBody"
    data-name="page-body"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-top-medium StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--padding-bottom-xlarge"
      style="height: 100%;"
    >
      <div
        class="Layout--horizontal StyledProps--main sloFiltersRow1"
      >
        <div
          class="StyledProps--font StyledProps--main"
        >
          <div
            class="Layout--horizontal StyledProps--main sloFilters"
          >
            <div
              class="Layout--vertical StyledProps--main StyledProps--margin-right-small"
              data-testid="userJourney-filter"
              style="width: 240px;"
            >
              <div
                class="bp3-popover-wrapper Select--main bp3-fill"
              >
                <div
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      placeholder="- Select -"
                      style="padding-right: 0px;"
                      type="text"
                      value="cv.slos.userJourney: all"
                    />
                    <span
                      class="bp3-input-action"
                    >
                      <span
                        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Layout--vertical StyledProps--main StyledProps--margin-right-small"
              data-testid="monitoredServices-filter"
              style="width: 240px;"
            >
              <div
                class="bp3-popover-wrapper Select--main bp3-fill"
              >
                <div
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      placeholder="- Select -"
                      style="padding-right: 0px;"
                      type="text"
                      value="cv.monitoredServices.title: all"
                    />
                    <span
                      class="bp3-input-action"
                    >
                      <span
                        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Layout--vertical StyledProps--main StyledProps--margin-right-small"
              data-testid="sloTargetAndBudget-filter"
              style="width: 240px;"
            >
              <div
                class="bp3-popover-wrapper Select--main bp3-fill"
              >
                <div
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      placeholder="- Select -"
                      style="padding-right: 0px;"
                      type="text"
                      value="cv.slos.sloTargetAndBudget.periodType: all"
                    />
                    <span
                      class="bp3-input-action"
                    >
                      <span
                        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Layout--vertical StyledProps--main StyledProps--margin-right-small"
              data-testid="sliType-filter"
              style="width: 240px;"
            >
              <div
                class="bp3-popover-wrapper Select--main bp3-fill"
              >
                <div
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      placeholder="- Select -"
                      style="padding-right: 0px;"
                      type="text"
                      value="cv.slos.sliType: all"
                    />
                    <span
                      class="bp3-input-action"
                    >
                      <span
                        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="bp3-card bp3-interactive bp3-elevation-0 Card--card sloRiskFilterCard Card--interactive"
          data-index="0"
          tabindex="0"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-form-help"
          >
            cv.slos.totalServices
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-variation-h2"
          >
            3
          </p>
        </div>
        <div
          class="bp3-card bp3-interactive bp3-elevation-0 Card--card sloRiskFilterCard Card--interactive"
          data-index="1"
          tabindex="0"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-form-help"
          >
            Healthy
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-green500 StyledProps--font-variation-h2"
          >
            2
          </p>
        </div>
      </div>
      <hr />
      <div
        class="sloCardContainer"
        data-testid="slo-card-container"
      >
        <div
          class="bp3-card bp3-elevation-0 Card--card sloCard"
        >
          <span
            data-testid="slo-card-header"
          >
            <button>
              Delete
            </button>
          </span>
          <span
            data-testid="slo-card-content"
          />
        </div>
      </div>
      <span
        style="flex-grow: 1;"
      />
      <div
        class="Layout--horizontal StyledProps--main Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
      >
        <span
          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
        >
          1
           of 
          1
        </span>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonLeft Button--with-current-color Button--withLeftIcon Button--size Button--small"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-arrow-left StyledProps--main StyledProps--padding-right-xsmall"
              icon="arrow-left"
            >
              <svg
                data-icon="arrow-left"
                height="12"
                viewBox="0 0 16 16"
                width="12"
              >
                <desc>
                  arrow-left
                </desc>
                <path
                  d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp3-button-text"
            >
              Prev
            </span>
          </button>
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--selected Button--with-current-color"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              1
            </span>
          </button>
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonRight Button--with-current-color Button--withRightIcon Button--size Button--small"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Next
            </span>
            <span
              class="bp3-icon bp3-icon-arrow-right StyledProps--main"
              icon="arrow-right"
            >
              <svg
                data-icon="arrow-right"
                height="12"
                viewBox="0 0 16 16"
                width="12"
              >
                <desc>
                  arrow-right
                </desc>
                <path
                  d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
        <span>
          Showing 4 per page
        </span>
      </div>
    </div>
  </div>
</div>
`;
